### PR TITLE
Added implementations - max_flow_bfs ,max_flow_dfs, knapsack, longest_palindromic_subsequence

### PR DESCRIPTION
**Pull request by saketrule**
Added a few implementations. 
Also changed corresponding README .md index

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,17 +65,21 @@ List of Implementations:
 │   ├── sudoku_solver.py
 │   └── walls_and_gates.py
 ├── dp
+|   ├── knapsack.py
```
Comment (keon) [inline README.md]: Seems like we have a conflict with current README.

File: dp/knapsack.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'''
+Given a list of tuples(w,v) lst, and maximum weight - bag,
+	w - weight
+	v - value
+of items, knapsack(lst) returns a solution to the Knapsack problem,
+i.e it returns a list of items which maximize the value within the maximum weight constraint.
+'''
+
+def knapsack(lst,bag):
+	def get_knapsack(lst,bag,table):
```
Comment (keon) [inline dp/knapsack.py]: applies to all code: please use 4 spaces :)

File: dp/knapsack.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+		return max_v
+	table = {}
+	val = get_knapsack(lst,bag,table)
+	sol = []
+	while(str(bag) in table and table[str(bag)][0]>0):
+		sol.append(table[str(bag)][1])
+		bag -= table[str(bag)][1][0]
+	return (val,sol)
+
+print(knapsack([(3,5),(2,1)],8))
```
Comment (keon) [inline dp/knapsack.py]: It'd be better to have this wrapped with `if __name__ `

**Comment by christianbender** [comment]
@goswami-rahul @keon  Should we close this PR? Because out of date.

**Comment by goswami-rahul** [comment]
yeah, i think it should be closed. There are some old conflicts.

**Pull request closed without merging**